Adding git version information to Site Info.

Delawen opened the pull request:
Also, check you are at least an authenticated user to ask the API about the site info.
![photo5971851446552996356](https://user-images.githubusercontent.com/726590/31332477-d68729ec-ace6-11e7-89a2-bd2ae5bd1fec.jpg)

So for each GeoNetwork build we can know the exact build it was compiled with, the commit message, how far is the commit from last version (substitutes SNAPSHOT with count of commits since tag), timestamp of creation and even if the source code was dirty on the moment of build (it means: if it had any modifications compared to the git source code).

fxprunayre commented:
We already have some of those info in 
![image](https://user-images.githubusercontent.com/1701393/31332708-c02e40f8-ace7-11e7-8c44-5b0cda4f015d.png)
at least commit number.

Delawen commented:
How is this generated? Is this shown on some UI? I see more useful info on my implementation but maybe it is because that implementation does not show everything.

juanluisrp commented:
Too many properties for me. I'd keep these:
* `git.tags`
* `git.commit.id.abbrev`
* `git.branch`

Delawen commented:
I see the other build api gets the info from config-spring-env.xml:


  <bean id="systemInfo" class="org.fao.geonet.SystemInfo">
    <constructor-arg value="${stagingProfile}"/>
    <constructor-arg value="${build.timestamp}"/>
    <constructor-arg value="${geonetwork.version}"/>
    <constructor-arg value="${geonetwork.subversion}"/>
    <constructor-arg value="${os.name} ${os.version} ${os.arch}"/>
    <constructor-arg value="${java.version}"/>
    <constructor-arg value="${java.vm.vendor}"/>
    <constructor-arg value="${parent.scm.url}"/>
    <constructor-arg value="${buildNumber}"/>
  </bean>

Delawen commented:
We can select what properties we want to generate on the git-commit-id-plugin in web/pom.xml.

Delawen commented:
I would substitute the buildnumber-maven-plugin with the git-commit-id-plugin as it has more details that can be configured if needed. Any comments against this? @fxprunayre ? I can inject this properties on the info/build api.

fxprunayre commented:
> I can inject this properties on the info/build api.

sounds good.

Delawen commented:
![image](https://user-images.githubusercontent.com/726590/31336966-8d84537e-acf9-11e7-96e3-cddf2ff96866.png)

Delawen commented:
I added the full commit hash as it was required to be placed on the MANIFEST.MF file.

Also added the inferred version because I think it makes sense as a measure of how far are you from the latest tagged version.

cmangeat commented on services/src/main/java/org/fao/geonet/api/site/SiteInformation.java:
```diff
@@ -200,13 +210,44 @@ private void loadDatabaseInfo(ServiceContext context) throws SQLException {
+        InputStream input = null;
+
+        try {
+            input = new FileInputStream(new File(getClass().getResource("/git.properties").getPath()));
```
Maybe getResourceAsStream ?

cmangeat commented on services/src/main/java/org/fao/geonet/api/site/SiteInformation.java:
```diff
@@ -200,13 +210,44 @@ private void loadDatabaseInfo(ServiceContext context) throws SQLException {
+        Properties prop = new Properties();
+        InputStream input = null;
+
+        try {
```
Why not a "try-with-resources" ?

cmangeat commented on services/src/main/java/org/fao/geonet/api/site/SiteInformation.java:
```diff
@@ -89,24 +93,37 @@ public void setDatabaseProperties(HashMap<String, String> databaseProperties) {
     }
 
+    @JsonProperty(value = "version")
+    public HashMap<String, String> getVersionProperties() {
```
Does really need properties to be a HashMap ? Is Map sufficient ? Is this case do we need to recopy around line 234 ?

josegar74 commented:
@Delawen I'm a bit confused about this PR, seem most of the files changed are not really related to the scope of the PR or am I confused?

Delawen commented:
Yes, I did a push force I shouldn't, fixing it... wait for it...

Delawen commented:
Done.
